cvmfs/swissknife_pull.cc

```diff
@@ -536,6 +536,7 @@ int swissknife::CommandPull::Main(const swissknife::ArgumentList &args) {
 
   if (!this->InitSignatureManager(master_keys, trusted_certs)) {
     LogCvmfs(kLogCvmfs, kLogStderr, "failed to initalize CVMFS signatures");
+    free(workers);
```

Comment:
Here I think a better fix is to initialize `workers` first to NULL and to move the actual memory allocation for `workers` down to where it is needed.